src/Common/SensitiveDataMasker.cpp

```diff
@@ -202,8 +202,13 @@ std::string wipeSensitiveDataAndCutToLength(const std::string & str, size_t max_
     if (auto * masker = SensitiveDataMasker::getInstance())
         masker->wipeSensitiveData(res);
 
-    if (max_length && (res.length() > max_length))
+    size_t length = res.length();
+    if (max_length && (length > max_length))
+    {
+        size_t truncated_len = length - max_length;
         res.resize(max_length);
+        res += "... (truncated " + std::to_string(truncated_len) + " chars)";
```

Comment:
But the length would be large then max_length in the end. If it strict rule and some one call such truncation once more, it will truncate the addition record about previous truncation.

Reply:
Not sure if it's really important, I suppose this function is always called once just before logging the query. Anyway, I updated the code it's a bit more complex, check out latest commit.

Reply:
I think both variants are ok, let's keep the current implementation.